src/components/common/transactions-list/transactions-list.component.coffee

```diff
@@ -141,21 +165,23 @@ module.component('transactionsList', {
 
     # If the transaction is a "child" forecast, we remove it from the list and trigger the callback
     # If it is a "parent" forecast, we remove the all group
-    ctrl.deleteTransaction = (trx) ->
+    deleteTransaction = (trx) ->
       _.remove(ctrl.currentAttributes.transactions, (trxInList) -> trxInList.id == trx.id)
       if trx.recurring_parent
         ctrl.onDeleteTransaction({ resourcesType: ctrl.resourcesType, trxId: trx.id })
       else
         # TODO: rework logic
-        ctrl.deleteTransactionsGroup({ recurring_parent: trx.id })
+        deleteTransactionsGroup({ recurring_parent: trx.id })
 
-    ctrl.deleteTransactionsGroup = (trx) ->
+    deleteTransactionsGroup = (trx) ->
       # TODO: should be accessible by recurring transactions only
-      return unless trx.recurring_parent
-      ctrl.onDeleteParentTransaction({ resourcesType: ctrl.resourcesType, trxId: trx.recurring_parent })
-      # Remove all children transactions and parent transaction if it is a forecast
-      _.remove(ctrl.currentAttributes.transactions, (trxInList) ->
-        (trxInList.recurring_parent == trx.recurring_parent) || (trxInList.id == trx.recurring_parent && trxInList.status == 'FORECAST')
+      return $q.when(null) unless trx.recurring_parent
+      args = { resourcesType: ctrl.resourcesType, trxId: trx.recurring_parent }
+      ctrl.onDeleteParentTransaction(args).finally(->
+        # Remove all children transactions and parent transaction if it is a forecast
```

Comment:
I wouldn't add a `finally` there: we don't need to wait for the API response to remove the elements from the list